core/application/tests/technicalImpl/dbModels/BloodDonationModel.test.ts

```diff
@@ -32,9 +142,66 @@ describe('BloodDonationModel', () => {
   })
 
   describe('getIndex', () => {
+    beforeEach(() => {
+      jest.restoreAllMocks()
+    })
+
     it('should return undefined for an unknown index', () => {
+      const indexDefinitions = {}
+      jest.spyOn(bloodDonationModel, 'getIndexDefinitions').mockReturnValue(indexDefinitions)
+
       const result = bloodDonationModel.getIndex('GSI', 'unknownIndex')
       expect(result).toBeUndefined()
     })
+
+    it('should return undefined when index type does not exist', () => {
+      const indexDefinitions = { GSI: {} }
+      jest.spyOn(bloodDonationModel, 'getIndexDefinitions').mockReturnValue(indexDefinitions)
+
+      const result = bloodDonationModel.getIndex('LSI', 'anyIndex')
+      expect(result).toBeUndefined()
+    })
+
+    it('should handle empty index definitions', () => {
+      const indexDefinitions = {}
+      jest.spyOn(bloodDonationModel, 'getIndexDefinitions').mockReturnValue(indexDefinitions)
+
+      const result = bloodDonationModel.getIndex('GSI', 'anyIndex')
+      expect(result).toBeUndefined()
+    })
+
+    it('should return undefined for different index types', () => {
```

Comment:
should this be a test.each

Reply:
Right vai! updated at 2206d1175c8d752e674ee8aeaf87992f540a312c